front/src/context/GlobalContext.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { useState, createContext } from 'react';
+import { ThemeProvider } from 'styled-components';
+import { theme } from '@/lib/styles';
+
+type ModeType = 'light' | 'dark';
+type ContextType = [ModeType, () => void];
+interface ChildrenType {
+  children: React.ReactNode;
+}
+
+export const GlobalContext = createContext<ContextType>({} as ContextType);
+
+function GlobalProvider({ children }: ChildrenType) {
+  const initialState = (window?.localStorage?.getItem('MODE') as ModeType) || 'light';
+
+  const [mode, setMode] = useState<ModeType>(initialState);
+  const toggleMode = () => {
+    const changedMode = mode === 'light' ? 'dark' : 'light';
+    window.localStorage.setItem('MODE', changedMode);
+    setMode(changedMode);
+  };
+
+  return (
+    <GlobalContext.Provider value={[mode, toggleMode]}>
+      <ThemeProvider theme={theme[mode]}>{children}</ThemeProvider>
+    </GlobalContext.Provider>
+  );
+}
+
```

Comment:
```suggestion
export function useThemeContext() {
  const value = useContext(GlobalContext);
  if (!value ) throw new Error('Cannot find GlobalContext');
  return value;
}
```
커스텀 훅 만들어서 사용하는 것도 좋다고 생각합니다.
```js
 const [mode, setMode] = useContext(GlobalContext);
// 위에 보다 아래 훅을 사용하는 게 더 가독성이 좋다고 생각합니다.
// 또 사용하는 컴포넌트에서 useContext를 임포트 안해도 되니까 더 깔끔하고요.
const [mode, setMode] = useThemeContext()
```
컨텍스트 파일 이름은 글로벌보다 테마컨텍스트가 더 어울리지 않나 생각합니다..!

Reply:
수정했습니다! 피드백 감사드립니다!